runner/src/server/plugins/engine/plugin.ts

```diff
@@ -186,9 +187,63 @@ export const plugin = {
       },
     });
 
+    const queryParamPreHandler = async (
+      request: HapiRequest,
+      h: HapiResponseToolkit
+    ) => {
+      const { query } = request;
+      const { id } = request.params;
+      const model = forms[id];
+      if (!model) {
+        throw Boom.notFound("No form found for id");
+      }
+
+      const prePopFields = model.fieldsForPrePopulation;
+      if (
+        Object.keys(query).length > 0 &&
+        Object.keys(prePopFields).length > 0
+      ) {
+        const { cacheService } = request.services([]);
+        const state = await cacheService.getState(request);
+        let newValues = {};
+        Object.entries(query).forEach(([key, value]) => {
+          console.log("Value of the query param: ", value);
+          if (reach(prePopFields, key) !== undefined) {
+            const keySplit = key.split(".");
+            if (keySplit.length === 1) {
+              const result = prePopFields[key].validate(value);
+              if (!result.error && !state[key]) {
+                newValues[key] = value;
+              }
+            } else {
+              const result = prePopFields[keySplit[0]][keySplit[1]].validate(
+                value
+              );
+              if (!result.error && !state[keySplit[0]]?.[keySplit[1]]) {
+                newValues[keySplit[0]] = {
+                  ...(newValues[keySplit[0]] ?? {}),
+                  [keySplit[1]]: value,
+                };
```

Comment:
hoek.reach should get the values without having to split sections/keys. Why do we need to do this manually?